src/app/(marketing)/profile/_components/EditContributions.tsx

```diff
@@ -0,0 +1,61 @@
+"use client"
+import { Button } from "@/components/ui/Button"
+import {
+  Dialog,
+  DialogBody,
+  DialogCloseButton,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/Dialog"
+// import { profileSchema } from "@/lib/validations/profileSchema"
+import { Schemas } from "@guildxyz/types"
+// import { zodResolver } from "@hookform/resolvers/zod"
+import { Pencil } from "@phosphor-icons/react"
+import { FormProvider, useForm } from "react-hook-form"
+
+export const EditContributions = (
+  contribution: Schemas["ProfileContributionUpdate"]
+) => {
+  const form = useForm<Schemas["ProfileContributionUpdate"]>({
+    // resolver: zodResolver(profileSchema),
+    defaultValues: {
+      ...contribution,
+    },
+    mode: "onTouched",
+  })
+
+  async function onSubmit(values: Schemas["ProfileContributionUpdate"]) {
+    console.log("edit contributions submit", values)
+  }
+
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        <Button variant="solid" size="icon">
+          <Pencil weight="bold" />
+        </Button>
+      </DialogTrigger>
+      <DialogContent size="lg" className="bg-background">
+        <DialogHeader>
+          <DialogTitle>Edit profile</DialogTitle>
+          <DialogCloseButton />
+        </DialogHeader>
+        <FormProvider {...form}>
+          <form onSubmit={form.handleSubmit(onSubmit)}>
+            <DialogBody>
```

Comment:
Will you implement the modal in this PR or you'll add it later in another one?

Reply:
The presentation part will be implemented here, presumably tomorrow.